src/monattack.cpp

```diff
@@ -1650,6 +1650,64 @@ void mattack::leap(monster *z, int index)
     }
 }
 
+void mattack::impale(monster *z, int index)
+{
+    if( !z->can_act() ) {
+        return;
+    }
+    Creature *target = z->attack_target();
+    if( target == nullptr || rl_dist( z->pos(), target->pos() ) > 1 ) {
+        return;
+    }
+
+    player *foe = dynamic_cast< player* >( target );
+    bool seen = g->u.sees( *z );
+
+    z->reset_special(index); // Reset timer
+    z->moves -= 80;
```

Comment:
The attack should take up more time(around 160 moves) because the zombie has to commit enough to puncture through armor.  Just one of these attacks landing will ruin someone's day.

Alternatively, you could subtract an additional 80 moves upon a successful hit, reflecting the time needed to withdraw the spike.